src/types/UiaTextRangeBase.cpp

```diff
@@ -985,28 +985,11 @@ std::wstring UiaTextRangeBase::_getTextValue(til::CoordType maxLength) const
         auto inclusiveEnd = _end;
         bufferSize.DecrementInBounds(inclusiveEnd, true);
 
-        // reserve size in accordance to extracted text
-        const auto textRects = buffer.GetTextRects(_start, inclusiveEnd, _blockRange, true);
-        const auto bufferData = buffer.GetText(true,
-                                               false,
-                                               textRects);
-        const size_t textDataSize = bufferData.text.size() * bufferSize.Width();
-        textData.reserve(textDataSize);
-        for (const auto& text : bufferData.text)
+        const auto req = TextBuffer::CopyRequest{ buffer, _start, inclusiveEnd, _blockRange, true, false, false, true };
+        auto plainText = buffer.GetPlainText(req);
+        if (plainText.size() <= maxLengthAsSize)

```

Comment:
ah, the code that is being removed reports to UIA _up to_ `maxLengthAsSize` whereas this will report nothing if the text is too long.

Reply:
(this was my one blocking comment!)

Reply:
Can you recheck with the latest changes?